server/lib/chat/chat-models.ts

```diff
@@ -578,3 +579,94 @@ export async function findChannelByName(
     done()
   }
 }
+
+/** Returns the number of channels that match a particular condition. */
+async function getChannelsCount(
+  whereCondition?: SQLStatement,
+  withClient?: DbClient,
+): Promise<number> {
+  const { client, done } = await db(withClient)
+  try {
+    const query = sql`
+      SELECT COUNT(*) AS count
+      FROM channels
+    `
+
+    if (whereCondition) {
+      query.append(whereCondition)
+    }
+
+    const result = await client.query<{ count: string }>(query)
+
+    return Number(result.rows[0].count)
+  } finally {
+    done()
+  }
+}
+
+/**
+ * Gets a list of chat channels for a particular user, optionally filtered by a `searchStr`. If the
+ * user is an admin, this method will return a full list of channels. Otherwise, this method will
+ * only return non-private channels and channels that the user is not joined in.
+ */
+export async function getChannelsForUser(
+  userId: SbUserId,
+  isAdmin: boolean,
+  limit: number,
+  pageNumber: number,
+  searchStr?: string,
```

Comment:
This many params should really go into an object, not passed positionally (especially true given that one of them is optional and you *also* have an optional client param, and specifying one doesn't mean you'd want to specify the other). [In case it's not clear based on all our other code, the `withClient` param should stay outside of the parameters object, as the second param]